app/preference.go

```diff
@@ -54,9 +54,11 @@ func (a *App) UpdatePreferences(userId string, preferences model.Preferences) *m
 	}
 
 	if err := a.Srv().Store.Channel().UpdateSidebarChannelsByPreferences(&preferences); err != nil {
-		return err
+		return model.NewAppError("DeletePreferences", "api.preference.update_preferences.update_sidebar.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
+
 	message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_SIDEBAR_CATEGORY_UPDATED, "", "", userId, nil)
+	// TODO this needs to be updated to include information on which categories changed
```

Comment:
We'll be addressing these TODOs shortly since Nevy is working on adding the client-side websocket handling right now for the new sidebar events.

Reply:
@hmhealey will this ever be done... is causing problems with v2, we are dealing with them, but a WS is being sent without any need for it.